src/fn/watch-project.js

```diff
@@ -0,0 +1,234 @@
+    }
+    if (configFileMatcher(fileName)) {
+        eventQueue.enqueue(async () => {
+            await compileAppSettings();
+            return fileName;
+        });
+        return true;
+    }
+    return false;
+};
+
+const watchProject = {
+    watch: async (api, blockFn, callback = {}, uploadStateOnStart = false) => {
+        if (uploadStateOnStart) {
+            await uploadInitialState(api);
+            info('Initial State Uploaded');
+        }
+
+        eventQueue = new Queue({ concurrent: 1, start: true });
+
+        fsEventsSubscription = await watcher.subscribe(environment.pathToProject, async (err, events) => {
+            if (err) {
+                error(err);
+                return;
+            }
+            for (const event of events) {
+                if (event.type !== 'update' && event.type !== 'create') {
+                    continue;
+                }
+                const changePath = event.path;
+                const parsedPath = path.parse(changePath);
+                const fileName = parsedPath.base;
+
+                if (parsedPath.dir === path.join(environment.pathToProject, APP_FORMS_PATH)) {
+                    processAppForm(fileName);
+                    continue;
+                }
+
+                if (parsedPath.dir.startsWith(path.join(environment.pathToProject, APP_FORMS_PATH))
+                    && path.parse(parsedPath.dir).dir.endsWith(APP_FORMS_PATH)) { // check if the directory's immediate parent is forms/app
+                    const dirName = path.parse(parsedPath.dir).base;
+                    processAppFormMedia(dirName, fileName);
+                    continue;
+                }
+
+                if (parsedPath.dir === path.join(environment.pathToProject, CONTACT_FORMS_PATH)) {
+                    processContactForm(fileName);
+                    continue;
+                }
+
+                if (parsedPath.dir === path.join(environment.pathToProject, COLLECT_FORMS_PATH)) {
+                    processCollectForm(fileName);
+                    continue;
+                }
+
+
+                if (parsedPath.dir === path.join(environment.pathToProject, TRANSLATIONS_DIR_PATH)) {
+                    await uploadCustomTranslations();
+                    if (callback) callback(fileName);
+                    continue;
+                }
+
+                if (parsedPath.dir === path.join(environment.pathToProject, APP_SETTINGS_DIR_PATH)) {
+                    processConfigFiles(api, fileName);
+                    continue;
+                }
+
+                if (parsedPath.dir === environment.pathToProject && parsedPath.base !== RESOURCE_CONFIG_PATH) {
+                    processConfigFiles(api, fileName);
+                    continue;
+                }
+
+                if (parsedPath.base === RESOURCE_CONFIG_PATH || parsedPath.dir === path.join(environment.pathToProject, RESOURCES_DIR_PATH)) {
+                    await uploadResources();
```

Comment:
We need to handle errors for all of these calls, right now they're unhandled rejections. 
For example, if I change the "resources.json" file and make the json invalid, I get: 
```
WARN Error parsing JSON in: /home/diana/projects/medic-utils/conf/default/resources.json 
(node:1952248) UnhandledPromiseRejectionWarning: SyntaxError: Unexpected token } in JSON at position 1502
    at JSON.parse (<anonymous>)
    at Object.readJson (/home/diana/projects/medic-conf/src/lib/sync-fs.js:29:17)
    at module.exports (/home/diana/projects/medic-conf/src/lib/upload-configuration-docs.js:47:19)
    at execute (/home/diana/projects/medic-conf/src/fn/upload-resources.js:19:12)
    at /home/diana/projects/medic-conf/src/fn/watch-project.js:206:27
(node:1952248) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:1952248) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```